packages/block-library/src/gallery/transforms.js

```diff
@@ -33,6 +34,33 @@ const parseShortcodeIds = ( ids ) => {
 	return ids.split( ',' ).map( ( id ) => parseInt( id, 10 ) );
 };
 
+function transformV1FormatFromThirdPartyBlocks( block ) {
+	const settings = select( blockEditorStore ).getSettings();
+	if (
+		settings.__unstableGalleryWithImageBlocks &&
+		block.name === 'core/gallery' &&
+		block.attributes?.images
+	) {
+		const innerBlocks = block.attributes.images.map( ( image ) => {
+			return createBlock( 'core/image', {
+				url: image.url,
+				id: image.id,
+			} );
+		} );
+
+		delete block.attributes.ids;
+		delete block.attributes.images;
+		block.innerBlocks = innerBlocks;
+	}
+
+	return block;
+}
+addFilter(
+	'blocks.switchToBlockType.transformedBlock',
+	'core.transformthirdpartygalleries',
```

Comment:
I can't find anything explicitly saying so, however looking at the source examples, it seems as if the common syntax for filter namespaces is something like `core/transform-third-party-galleries` or `core/gallery/transform-third-party`